### PR TITLE
Add more detailed validator information in debug page

### PR DESCRIPTION
**Pull request by mzhangmzz**
Screen Shot 2022-07-25 at 8 14 52 PM

### CODE REVIEW

File: chain/jsonrpc/res/epoch_info.html [hunk 1]
```diff
@@ -71,6 +71,48 @@
             return bytes.toFixed(dp) + ' ' + units[u];
         }
 
+        function process_validator_info(validator_info) {
+            let total_stake = 0;
+            validator_info.current_validators.forEach(validator => {
+                total_stake += parseInt(validator.stake);
+            })
+            $('.js-tbody-curepoch').append("Total stake " + total_stake)
```
Comment (mm-near) [inline chain/jsonrpc/res/epoch_info.html:79]: consider dividing total stake by 10^24 (to make it more readable)

File: chain/jsonrpc/res/epoch_info.html [hunk 2]
```diff
@@ -71,6 +71,48 @@
...
+            validator_info.current_validators.forEach(validator => {
+                let row = $('<tr>');
+                console.log(validator.stake + "validator stake\n");
```
Comment (mm-near) [inline chain/jsonrpc/res/epoch_info.html:82]: remove

File: chain/jsonrpc/res/epoch_info.html [hunk 3]
```diff
@@ -71,6 +71,48 @@
...
+                row.append($('<td>').append(validator.account_id));
+                row.append($('<td>').append((100 * validator.stake / total_stake).toFixed(2) + "%"));
+                row.append($('<td>').append(validator.shards.join(",")));
+                row.append($('<td>').append(validator.num_expected_blocks - validator.num_produced_blocks));
+                row.append($('<td>').append(validator.num_expected_blocks));
+                row.append($('<td>').append(validator.num_expected_chunks - validator.num_produced_chunks));
+                row.append($('<td>').append(validator.num_expected_chunks));
```
Comment (mm-near) [inline chain/jsonrpc/res/epoch_info.html:89]: let's add a column with the percent, that we're using in prod for kickout (num_producer / num_expected).

File: chain/jsonrpc/res/epoch_info.html [hunk 4]
```diff
@@ -71,6 +71,48 @@
...
+                row1.append($('<td>').append(proposal.account_id));
+                row2.append($('<td>').append(proposal.stake + "/" + proposal.stake/total_stake.toFixed(2) + "%"));
+            })
+            $('.js-tbody-curepoch-proposals').append(row1).append(row2);
+
+            row1 = $('<tr>');
+            row2 = $('<tr>');
+            row1.append("<th> Account Id </th>");
+            row2.append("<th> Kickout Reason </th>");
+            validator_info.prev_epoch_kickout.forEach(kickout=> {
```
Comment (mm-near) [inline chain/jsonrpc/res/epoch_info.html:107]: this can get very wide.. (as you're adding a column per each validator that got kicked out, right?)

Comment (mm-near) [reply chain/jsonrpc/res/epoch_info.html:107]: maybe let's make it still a 'one validator per row' -- but  in such a way that we can 'fold' this whole section.

**Review by mm-near — APPROVED** [review]

File: chain/jsonrpc/res/epoch_info.html [hunk 5]
```diff
@@ -208,12 +256,39 @@ <h3><span class="js-next-epoch"></span></h3>
                 <th>Protocol version</th>
                 <th>First block</th>
                 <th>Epoch start</th>
+                <th>Block producers</th>
+                <th>Chunk only producers</th>
```
Comment (mm-near) [inline chain/jsonrpc/res/epoch_info.html:260]: oh - and we could also compute the number of block and chunks missed during the whole epoch, right?

Comment (mm-near) [reply chain/jsonrpc/res/epoch_info.html:260]: uh - unfortunately we cannot do that..

As EpochInfo (for some reason) keeps only the stats from the producers that were not kicked out:
        /// Only for validators who met the threshold and didn't get slashed
        pub validator_block_chunk_stats: HashMap<AccountId, BlockChunkValidatorStats>,

Comment (mzhangmzz) [reply chain/jsonrpc/res/epoch_info.html:260]: We could, if we record this information somewhere. It's not too difficult to compute. I can do that in another PR.

**Pull request merged**